brew typecheck not working on ARM

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

<!-- To help us debug your issue, please complete these sections: -->

## What you were trying to do (and why)

Run `brew man` and `brew typecheck` as part of a PR.

## What happened (include command output)

<!-- replace me -->

<details>
  <summary>Command output</summary>
  <pre>
 $ brew man --debug
Writing markdown to /opt/homebrew/docs/Manpage.md
Error: Broken pipe
/opt/homebrew/Library/Homebrew/dev-cmd/man.rb:131:in `write'
/opt/homebrew/Library/Homebrew/dev-cmd/man.rb:131:in `block in convert_man_page'
/opt/homebrew/Library/Homebrew/utils/popen.rb:52:in `block in popen'
/opt/homebrew/Library/Homebrew/utils/popen.rb:48:in `popen'
/opt/homebrew/Library/Homebrew/utils/popen.rb:48:in `popen'
/opt/homebrew/Library/Homebrew/dev-cmd/man.rb:130:in `convert_man_page'
/opt/homebrew/Library/Homebrew/dev-cmd/man.rb:62:in `regenerate_man_pages'
/opt/homebrew/Library/Homebrew/dev-cmd/man.rb:44:in `man'
/opt/homebrew/Library/Homebrew/brew.rb:124:in `<main>'

$ brew typecheck --debug
/opt/homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/sorbet-0.5.6189/bin/srb: line 46: 99392 Illegal instruction: 4  "${sorbet}" "${args[@]}"
Check https://docs.brew.sh/Typechecking for more information on how to resolve these errors.
  </pre>
</details>

## What you expected to happen

The commands to run and not error. 
## Step-by-step reproduction instructions (by running `brew` commands)

`brew man` or `brew typecheck`

## Output of `brew config` and `brew doctor` commands

<pre>
$ brew config
HOMEBREW_VERSION: 2.7.1-119-g4a90a5a
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4a90a5a5d6210b3ab526d68c3c9330c91bc4a55d
Last commit: 29 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 39af1529f356db9c9b20409e24ed763d8950f960
Core tap last commit: 63 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DEVELOPER: set
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ANALYTICS: set
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.30.0 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-arm64
CLT: 12.3.0.0.1.1607026830
Xcode: N/A
Rosetta 2: false
$ brew doctor
Your system is ready to brew.
</pre>

<details>
 <summary>universal-darwin-20/2.6.0/rdiscount-2.2.0.2/mkmf.log for system ruby</summary>

```
have_func: checking for random()... -------------------- yes

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))random; return !p; }
/* end */

--------------------

have_func: checking for srandom()... -------------------- yes

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))srandom; return !p; }
/* end */

--------------------

have_func: checking for rand()... -------------------- yes

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))rand; return !p; }
/* end */

--------------------

have_func: checking for srand()... -------------------- yes

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))srand; return !p; }
/* end */

--------------------

check_sizeof: checking size of unsigned long... -------------------- 8

"xcrun clang -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC   -c conftest.c"
conftest.c:7:20: error: 'conftest_const' declared as an array with a negative size
int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: typedef unsigned long rbcv_typedef_;
4: rbcv_typedef_ *rbcv_ptr_;
5:
6: /*top*/
7: int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
/* end */

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: typedef unsigned long rbcv_typedef_;
 4: rbcv_typedef_ *rbcv_ptr_;
 5:
 6: #include <stdio.h>
 7: /*top*/
 8: typedef unsigned
 9: #ifdef PRI_LL_PREFIX
10: #define PRI_CONFTEST_PREFIX PRI_LL_PREFIX
11: LONG_LONG
12: #else
13: #define PRI_CONFTEST_PREFIX "l"
14: long
15: #endif
16: conftest_type;
17: conftest_type conftest_const = (conftest_type)(sizeof((*rbcv_ptr_)));
18: int main() {printf("%"PRI_CONFTEST_PREFIX"u\n", conftest_const); return 0;}
/* end */

./conftest |
--------------------

check_sizeof: checking size of unsigned int... -------------------- 4

"xcrun clang -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC   -c conftest.c"
conftest.c:7:20: error: 'conftest_const' declared as an array with a negative size
int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: typedef unsigned int rbcv_typedef_;
4: rbcv_typedef_ *rbcv_ptr_;
5:
6: /*top*/
7: int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
/* end */

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: typedef unsigned int rbcv_typedef_;
 4: rbcv_typedef_ *rbcv_ptr_;
 5:
 6: #include <stdio.h>
 7: /*top*/
 8: typedef unsigned
 9: #ifdef PRI_LL_PREFIX
10: #define PRI_CONFTEST_PREFIX PRI_LL_PREFIX
11: LONG_LONG
12: #else
13: #define PRI_CONFTEST_PREFIX "l"
14: long
15: #endif
16: conftest_type;
17: conftest_type conftest_const = (conftest_type)(sizeof((*rbcv_ptr_)));
18: int main() {printf("%"PRI_CONFTEST_PREFIX"u\n", conftest_const); return 0;}
/* end */

./conftest |
--------------------

check_sizeof: checking size of unsigned int... -------------------- 4

"xcrun clang -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC   -c conftest.c"
conftest.c:7:20: error: 'conftest_const' declared as an array with a negative size
int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: typedef unsigned int rbcv_typedef_;
4: rbcv_typedef_ *rbcv_ptr_;
5:
6: /*top*/
7: int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
/* end */

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: typedef unsigned int rbcv_typedef_;
 4: rbcv_typedef_ *rbcv_ptr_;
 5:
 6: #include <stdio.h>
 7: /*top*/
 8: typedef unsigned
 9: #ifdef PRI_LL_PREFIX
10: #define PRI_CONFTEST_PREFIX PRI_LL_PREFIX
11: LONG_LONG
12: #else
13: #define PRI_CONFTEST_PREFIX "l"
14: long
15: #endif
16: conftest_type;
17: conftest_type conftest_const = (conftest_type)(sizeof((*rbcv_ptr_)));
18: int main() {printf("%"PRI_CONFTEST_PREFIX"u\n", conftest_const); return 0;}
/* end */

./conftest |
--------------------

check_sizeof: checking size of unsigned short... -------------------- 2

"xcrun clang -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC   -c conftest.c"
conftest.c:7:20: error: 'conftest_const' declared as an array with a negative size
int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: typedef unsigned short rbcv_typedef_;
4: rbcv_typedef_ *rbcv_ptr_;
5:
6: /*top*/
7: int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
/* end */

"xcrun clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/universal-darwin20 -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0/ruby/backward -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/include/ruby-2.6.0 -I. -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT    -g -Os -pipe -DHAVE_GCC_ATOMIC_BUILTINS -DUSE_FFI_CLOSURE_ALLOC conftest.c  -L. -L/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib     -lruby.2.6   "
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.2.Internal.sdk/usr/local/lib'
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: typedef unsigned short rbcv_typedef_;
 4: rbcv_typedef_ *rbcv_ptr_;
 5:
 6: #include <stdio.h>
 7: /*top*/
 8: typedef unsigned
 9: #ifdef PRI_LL_PREFIX
10: #define PRI_CONFTEST_PREFIX PRI_LL_PREFIX
11: LONG_LONG
12: #else
13: #define PRI_CONFTEST_PREFIX "l"
14: long
15: #endif
16: conftest_type;
17: conftest_type conftest_const = (conftest_type)(sizeof((*rbcv_ptr_)));
18: int main() {printf("%"PRI_CONFTEST_PREFIX"u\n", conftest_const); return 0;}
/* end */

./conftest |
--------------------
```
</details>

<details>
 <summary>arm64-darwin-20/2.6.0-static/rdiscount-2.2.0.2/mkmf.log for portable ruby</summary>

```
have_func: checking for random()... -------------------- yes

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))random; return !p; }
/* end */

--------------------

have_func: checking for srandom()... -------------------- yes

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))srandom; return !p; }
/* end */

--------------------

have_func: checking for rand()... -------------------- yes

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))rand; return !p; }
/* end */

--------------------

have_func: checking for srand()... -------------------- yes

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     int (* volatile tp)(void)=(int (*)(void))&t;
 9:     printf("%d", (*tp)());
10:   }
11:
12:   return 0;
13: }
14: int t(void) { void ((*volatile p)()); p = (void ((*)()))srand; return !p; }
/* end */

--------------------

check_sizeof: checking size of unsigned long... -------------------- 8

"clang -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe   -c conftest.c"
conftest.c:7:20: error: 'conftest_const' declared as an array with a negative size
int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: typedef unsigned long rbcv_typedef_;
4: rbcv_typedef_ *rbcv_ptr_;
5:
6: /*top*/
7: int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
/* end */

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: typedef unsigned long rbcv_typedef_;
 4: rbcv_typedef_ *rbcv_ptr_;
 5:
 6: #include <stdio.h>
 7: /*top*/
 8: typedef unsigned
 9: #ifdef PRI_LL_PREFIX
10: #define PRI_CONFTEST_PREFIX PRI_LL_PREFIX
11: LONG_LONG
12: #else
13: #define PRI_CONFTEST_PREFIX "l"
14: long
15: #endif
16: conftest_type;
17: conftest_type conftest_const = (conftest_type)(sizeof((*rbcv_ptr_)));
18: int main() {printf("%"PRI_CONFTEST_PREFIX"u\n", conftest_const); return 0;}
/* end */

./conftest |
--------------------

check_sizeof: checking size of unsigned int... -------------------- 4

"clang -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe   -c conftest.c"
conftest.c:7:20: error: 'conftest_const' declared as an array with a negative size
int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: typedef unsigned int rbcv_typedef_;
4: rbcv_typedef_ *rbcv_ptr_;
5:
6: /*top*/
7: int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
/* end */

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: typedef unsigned int rbcv_typedef_;
 4: rbcv_typedef_ *rbcv_ptr_;
 5:
 6: #include <stdio.h>
 7: /*top*/
 8: typedef unsigned
 9: #ifdef PRI_LL_PREFIX
10: #define PRI_CONFTEST_PREFIX PRI_LL_PREFIX
11: LONG_LONG
12: #else
13: #define PRI_CONFTEST_PREFIX "l"
14: long
15: #endif
16: conftest_type;
17: conftest_type conftest_const = (conftest_type)(sizeof((*rbcv_ptr_)));
18: int main() {printf("%"PRI_CONFTEST_PREFIX"u\n", conftest_const); return 0;}
/* end */

./conftest |
--------------------

check_sizeof: checking size of unsigned int... -------------------- 4

"clang -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe   -c conftest.c"
conftest.c:7:20: error: 'conftest_const' declared as an array with a negative size
int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: typedef unsigned int rbcv_typedef_;
4: rbcv_typedef_ *rbcv_ptr_;
5:
6: /*top*/
7: int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
/* end */

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: typedef unsigned int rbcv_typedef_;
 4: rbcv_typedef_ *rbcv_ptr_;
 5:
 6: #include <stdio.h>
 7: /*top*/
 8: typedef unsigned
 9: #ifdef PRI_LL_PREFIX
10: #define PRI_CONFTEST_PREFIX PRI_LL_PREFIX
11: LONG_LONG
12: #else
13: #define PRI_CONFTEST_PREFIX "l"
14: long
15: #endif
16: conftest_type;
17: conftest_type conftest_const = (conftest_type)(sizeof((*rbcv_ptr_)));
18: int main() {printf("%"PRI_CONFTEST_PREFIX"u\n", conftest_const); return 0;}
/* end */

./conftest |
--------------------

check_sizeof: checking size of unsigned short... -------------------- 2

"clang -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe   -c conftest.c"
conftest.c:7:20: error: 'conftest_const' declared as an array with a negative size
int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: typedef unsigned short rbcv_typedef_;
4: rbcv_typedef_ *rbcv_ptr_;
5:
6: /*top*/
7: int conftest_const[(sizeof((*rbcv_ptr_)) < 0) ? 1 : -1];
/* end */

"clang -o conftest -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/arm64-darwin20 -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0/ruby/backward -I/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/include/ruby-2.6.0 -I.  -I/opt/homebrew/opt/portable-readline/include -I/opt/homebrew/opt/portable-libyaml/include -I/opt/homebrew/opt/portable-openssl/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -O3 -ggdb3 -Wall -Wextra -Wdeclaration-after-statement -Wdeprecated-declarations -Wdivision-by-zero -Wimplicit-function-declaration -Wimplicit-int -Wmisleading-indentation -Wpointer-arith -Wshorten-64-to-32 -Wwrite-strings -Wmissing-noreturn -Wno-constant-logical-operand -Wno-long-long -Wno-missing-field-initializers -Wno-overlength-strings -Wno-parentheses-equality -Wno-self-assign -Wno-tautological-compare -Wno-unused-parameter -Wno-unused-value -Wunused-variable -Wextra-tokens  -pipe conftest.c  -L. -L/opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib -L. -Wl,-search_paths_first -fstack-protector-strong -L/opt/homebrew/lib -L/opt/homebrew/opt/portable-readline/lib -L/opt/homebrew/opt/portable-libyaml/lib -L/opt/homebrew/opt/portable-openssl/lib     -lruby.2.6-static -framework Security -framework Foundation -lpthread -ldl -lobjc   "
checked program was:
/* begin */
 1: #include "ruby.h"
 2:
 3: typedef unsigned short rbcv_typedef_;
 4: rbcv_typedef_ *rbcv_ptr_;
 5:
 6: #include <stdio.h>
 7: /*top*/
 8: typedef unsigned
 9: #ifdef PRI_LL_PREFIX
10: #define PRI_CONFTEST_PREFIX PRI_LL_PREFIX
11: LONG_LONG
12: #else
13: #define PRI_CONFTEST_PREFIX "l"
14: long
15: #endif
16: conftest_type;
17: conftest_type conftest_const = (conftest_type)(sizeof((*rbcv_ptr_)));
18: int main() {printf("%"PRI_CONFTEST_PREFIX"u\n", conftest_const); return 0;}
/* end */

./conftest |
--------------------
```
<details>

--- a/Library/Homebrew/dev-cmd/typecheck.rb
+++ b/Library/Homebrew/dev-cmd/typecheck.rb
@@ -13,8 +13,6 @@ module Homebrew
     Homebrew::CLI::Parser.new do
       description <<~EOS
         Check for typechecking errors using Sorbet.
-
-        Not (yet) working on Apple Silicon.
       EOS
       switch "--fix",
              description: "Automatically fix type errors."
@@ -44,11 +42,6 @@ module Homebrew
 
   sig { void }
   def typecheck
-    # TODO: update description above if removing this.
-    if Hardware::CPU.arm? || Hardware::CPU.in_rosetta2?
-      raise UsageError, "not (yet) working on Apple Silicon or Rosetta 2!"
-    end
-
     args = typecheck_args.parse
 
     Homebrew.install_bundler_gems!(groups: ["sorbet"])

